ceno_host/src/lib.rs

```diff
@@ -127,8 +127,10 @@ impl CenoStdin {
     pub fn write(
         &mut self,
         item: &impl for<'a> Serialize<HighSerializer<AlignedVec, ArenaHandle<'a>, Error>>,
-    ) -> Result<(), Error> {
-        to_bytes::<Error>(item).map(|bytes| self.write_slice(bytes))
+    ) -> Result<&mut Self, Error> {
+        to_bytes::<Error>(item)
+            .map(|bytes| self.write_slice(bytes))
```

Comment:
or maybe `write_slice` returns `&mut Self` too?

Reply:
also, although that Result::map is not lazy, this code looks really _SUSPICIOUS_ to me.

Reply:
> or maybe `write_slice` returns `&mut Self` too?

Good idea.  That also removes the need to `map` the `Result`.